src/python/pants/backend/python/lint/yapf/rules_integration_test.py

```diff
@@ -0,0 +1,261 @@
+    lint_results, fmt_result = run_yapf(rule_runner, [tgt])
+    assert len(lint_results) == 1
+    assert lint_results[0].exit_code == 0
+    assert lint_results[0].stderr == ""
+    assert fmt_result.output == get_digest(rule_runner, {"f.py": GOOD_FILE})
+    assert fmt_result.did_change is False
+
+
+def test_failing_source(rule_runner: RuleRunner) -> None:
+    rule_runner.write_files({"f.py": BAD_FILE, "BUILD": "python_library(name='t')"})
+    tgt = rule_runner.get_target(Address("", target_name="t", relative_file_path="f.py"))
+    lint_results, fmt_result = run_yapf(rule_runner, [tgt])
+    assert len(lint_results) == 1
+    assert lint_results[0].exit_code == 1
+    assert all(msg in lint_results[0].stdout for msg in ("reformatted", "original"))
+    assert fmt_result.skipped is False
+    assert fmt_result.output == get_digest(rule_runner, {"f.py": FIXED_BAD_FILE})
+    assert fmt_result.did_change is True
+
+
+def test_multiple_targets(rule_runner: RuleRunner) -> None:
+    rule_runner.write_files(
+        {"good.py": GOOD_FILE, "bad.py": BAD_FILE, "BUILD": "python_library(name='t')"}
+    )
+    tgts = [
+        rule_runner.get_target(Address("", target_name="t", relative_file_path="good.py")),
+        rule_runner.get_target(Address("", target_name="t", relative_file_path="bad.py")),
+    ]
+    lint_results, fmt_result = run_yapf(rule_runner, tgts)
+    assert len(lint_results) == 1
+    assert lint_results[0].exit_code == 1
+    assert all(msg in lint_results[0].stdout for msg in ("reformatted", "original", "bad.py"))
+    assert "good.py" not in lint_results[0].stdout
+    assert fmt_result.output == get_digest(
+        rule_runner, {"good.py": GOOD_FILE, "bad.py": FIXED_BAD_FILE}
+    )
+    assert fmt_result.did_change is True
+
+
+@pytest.mark.parametrize(
+    "path,section,extra_args",
+    (
+        (".style.yapf", "style", []),
+        ("setup.cfg", "yapf", []),
+        ("pyproject.toml", "tool.yapf", []),
+        ("custom.style", "style", ["--yapf-config=custom.style"]),
+    ),
+)
+def test_config_file(
+    rule_runner: RuleRunner, path: str, section: str, extra_args: list[str]
+) -> None:
+    rule_runner.write_files(
+        {
+            "f.py": NEEDS_FORMATTING_FILE,
+            "BUILD": "python_library(name='t', interpreter_constraints=['==3.9.*'])",
+            path: f"[{section}]\nindent_width = 2\n",
+        }
+    )
+    tgt = rule_runner.get_target(Address("", target_name="t", relative_file_path="f.py"))
+    lint_results, fmt_result = run_yapf(rule_runner, [tgt], extra_args=extra_args)
+    assert len(lint_results) == 1
+    assert lint_results[0].exit_code == 1
+    assert all(msg in lint_results[0].stdout for msg in ("reformatted", "original", "f.py"))
+    assert fmt_result.output == get_digest(
+        rule_runner, {"f.py": FIXED_NEEDS_FORMATTING_FILE_INDENT2}
+    )
+    assert fmt_result.did_change is True
+
+
+def test_inline_style_overrides_config_file(rule_runner: RuleRunner) -> None:
```

Comment:
This feels more like a test for Yapf itself than for Pants. Imo, Pants shouldn't be super concerned about precedence orders for tools, all that is relevant to us is correctly invoking the tool like setting up argv and populating chroot.

Thoughts if there's something specific to Pants that is worth double checking here?

Reply:
Fully agree! Removed this one.